core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -60,13 +62,17 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
   override def tearDown() {
     consumer1.close()
     consumer2.close()
+    producers.foreach(_.close())
```

Comment:
It seems like almost all of the tests already close the producer (`testWrongSerializer` doesn't seem to do it). Is the idea that this is a fallback to avoid leaks? Should we remove the `close` from tests where `close` itself is not being tested?